### PR TITLE
Remediation and improvement for file_permissions_home_dirs rule

### PR DESCRIPTION
**Pull request by marcusburghardt**
Reviewing the issue https://github.com/ComplianceAsCode/content/issues/7890 it was noticed that this rule,
despite not being so used as the `file_permissions_home_directories`, was missing remediation and the OVAL
could be improved to be in alignment with other rules related to home directories owned by interactive users.

### CODE REVIEW

**Review by matejak — CHANGES_REQUESTED** [review]
The rule got improved by pulling the OVAL code from the `file_permissions_home_directories` rule, and basically ditto for the Bash and Ansible remediations.
The OVAL code duplication could possibly be reduced by using the `extend_definition` mechanics, but it is probably not worth it, as Bash and Ansible will be duplicated as well.

What about adding a deprecation notice to the rule that it is substituted by the `file_permissions_home_directories` one to the rule yaml and perhaps to remediations? Perhaps a `..._deprecated_by(rule_id)` macro could come in handy :-)

**Comment by jan-cerny** [comment]
@marcusburghardt ping :smile:

**Comment by marcusburghardt** [comment]
After long time, I finally managed to resume the work in this PR. My last commit introduced a new macro.
I would like more opinions about pros and cons informing a release number on it. The intention it to make it explicit `when` the deprecation was set, but I am not sure if the release number is the best way to provide this information.

File: shared/macros/warning.jinja [hunk 1]
```diff
@@ -12,6 +12,19 @@
...
 {{% endmacro %}}
 
+{{#
+    Includes a deprecation warning in rules which are no longer recommended in favor of a similar
+    and more updated rule.
+
+    :param rule: Rule which replaces the deprecated rule.
+    :param release: Release number since the rule is officially deprecated.
+#}}
+{{% macro warning_rule_deprecated_by(rule, release) %}}
```
Comment (matejak) [inline shared/macros/warning.jinja:22]:
```suggestion
{{% macro warning_rule_deprecated_by(rule, release="") %}}
```

File: shared/macros/warning.jinja [hunk 2]
```diff
@@ -12,6 +12,19 @@
...
+- general: |-
+    This rule is deprecated in favor of the <code>{{{ rule }}}</code> rule.
+    Please consider replacing this rule in your files as it is not expected to receive updates
+    as of version <code>{{{ release }}}</code>.
```
Comment (matejak) [inline shared/macros/warning.jinja:26]:
```suggestion
    Please consider replacing this rule in your files as it is not expected to receive
    {{%- if release %}}
    updates as of version <code>{{{ release }}}</code>.
    {{%- else %}}
    further updates.
    {{% endif -%}}
```

File: linux_os/guide/system/accounts/accounts-session/file_permissions_home_dirs/bash/shared.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# platform = multi_platform_all
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+
+for home_dir in $(awk -F':' '{ if ($4 >= {{{ uid_min }}} && $4 != 65534) print $6 }' /etc/passwd); do
```
Comment (matejak) [inline linux_os/guide/system/accounts/accounts-session/file_permissions_home_dirs/bash/shared.sh:7]: What is that `65534`? A max uid? Would it be possible to make an explanatory variable declared in the macro definitions file?

File: linux_os/guide/system/accounts/accounts-session/file_permissions_home_dirs/bash/shared.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    # Only update the permissions when necessary. This will avoid changing the inode timestamp when
+    # the permission is already defined as expected, therefore not impacting in possible integrity
+    # check systems that also check inodes timestamps.
+    find $home_dir -perm /7027 -exec chmod u-s,g-w-s,o=- {} \;
```
Comment (matejak) [inline linux_os/guide/system/accounts/accounts-session/file_permissions_home_dirs/bash/shared.sh:11]: Wouldn't it make sense to quote `$home_dir`?

File: linux_os/guide/system/accounts/accounts-session/file_permissions_home_dirs/bash/shared.sh (hunk 3 above)
Comment (marcusburghardt) [reply linux_os/guide/system/accounts/accounts-session/file_permissions_home_dirs/bash/shared.sh:7]: This uid is used in the special case of `nobody` user.

File: shared/macros/warning.jinja (hunk 2 above)
Comment (marcusburghardt) [reply shared/macros/warning.jinja:26]: Great. Thanks

**Comment by matejak** [comment]
Thank you, I like those changes very much.

**Pull request merged**